src/device/pci/config_space.rs

```diff
@@ -222,6 +222,36 @@ impl ConfigSpaceBuilder {
         self
     }
 
+    /// Add a Base Address Register (BAR) for a non-prefetchable 64-bit memory region.
+    ///
+    /// This is the typical BAR type for MMIO regions.
+    ///
+    /// Size must be a power of 2 and at least 16 bytes, but 4 KiB is the recommended minimum.
```

Comment:
I think we should follow Rust best practices for documentation

```suggestion
    /// # Arguments
    /// - `index`: ...
    /// - `size`: Size must be a power of 2 and at least 16 bytes, but 4 KiB is the recommended minimum.
```

Reply:
The whole function duplicates the existing 32 bit BAR logic and comment and is still missing tests. We can probably clean up the comment for both a bit.

Reply:
If the 64-bit BAR is not required to make progress, I'd like to remove it from this PR. The problems here are that it:

- doesn't handle BARs that are >4G
- has no tests

and of course it's not required at all right now.

Reply:
I reverted to the regular 32 bit bar. We still see the same access by the driver.